src/query/parser/promql/parse_test.go

```diff
@@ -541,6 +541,61 @@ func TestMissingTagsDoNotPanic(t *testing.T) {
 	assert.NotPanics(t, func() { _, _, _ = p.DAG() })
 }
 
+var functionArgumentExpressionTests = []struct {
+	name string
+	q    string
+}{
+	{
+		"scalar argument",
+		"vector(((1)))",
```

Comment:
I see that for whatever reason `unwrapParenExpr` does not recurse into the parenthesized expression.
Are you sure it would handle the nested `((1))` properly?

Reply:
It should loop and reassing `expr` until it's not `ParenExpr` anymore 

https://github.com/m3db/m3/blob/83fb11fa71dfd35e3642f85d2337b908ac31d2a5/src/query/parser/promql/matchers.go#L127-L135

Reply:
Oh, my bad, failed to see the `for`:)